tests/blackbox.py

```diff
@@ -46,12 +49,106 @@ class AwsTestConfigSetImpl(AwsTestConfig):
     name = 'aws+impl'
 
     # The same as AwsTestConfig but with a WALE_S3_ENDPOINT override.
-    def patch(self, test_name, default_test_bucket):
+    def patch(self, test_name):
         self.monkeypatch.setenv(
             'WALE_S3_ENDPOINT',
             'http+path://s3-us-west-1.amazonaws.com:80')
-        return AwsTestConfig.patch(self, test_name, default_test_bucket)
+        return AwsTestConfig.patch(self, test_name)
+
+
+class AwsInstanceProfileTestConfig(object):
+    name = 'aws+instance-profile'
+
+    def __init__(self, request):
+        self.request = request
+        self.monkeypatch = request.getfuncargvalue('monkeypatch')
+        self.default_test_bucket = request.getfuncargvalue(
+            'default_test_bucket')
+
+    def patch(self, test_name):
+        # Get STS-vended credentials to stand in for Instance Profile
+        # credentials before scrubbing the environment of AWS
+        # environment variables.
+        c = s3_integration_help.sts_conn()
+        policy = s3_integration_help.make_policy(self.default_test_bucket,
+                                                 test_name)
+        fed = c.get_federation_token(self.default_test_bucket, policy=policy)
+
+        # Scrub AWS environment-variable based cred to make sure the
+        # instance profile path is used.
+        for name in _AWS_CRED_ENV_VARS:
+            self.monkeypatch.delenv(name, raising=False)
+
+        self.monkeypatch.setenv('WALE_S3_PREFIX', 's3://{0}/{1}'
+                                .format(self.default_test_bucket, test_name))
+        self.monkeypatch.setenv('AWS_REGION', 'us-west-1')
+
+        # Patch boto.utils.get_instance_metadata to return a ginned up
+        # credential.
+        m = {
+            "Code": "Success",
+            "LastUpdated": "3014-01-11T02:13:53Z",
+            "Type": "AWS-HMAC",
+            "AccessKeyId": fed.credentials.access_key,
+            "SecretAccessKey": fed.credentials.secret_key,
+            "Token": fed.credentials.session_token,
+            "Expiration": "3014-01-11T08:16:59Z"
+        }
+
+        from boto import provider
```

Comment:
sometimes if you know that this code will only be used in this one instance, it's best to put it locally to namespace it, so the rest of the code can use the `provider` keyword (which is pretty generic).